src/main.c

```diff
@@ -4,143 +4,248 @@
+                
+                case CX_SC_Auto:
+                        printf("auto"); 
+                        break;
+
+                case CX_SC_Register:
+                        printf("register");
+                        break;
+        }
 }
 
 void enum_handler(CXCursor cursor, CXCursor parent, CXClientData client_data) 
 {
-  printf("EnumDecl\t\"");
-  print_cursor_spelling(cursor);
-  printf("\"\t\tParent = \"");
-  //print_parent(cursor);
-  print_cursor_spelling(parent);
-  printf("\"\n");
-  clang_visitChildren(cursor, enum_child_visitor, 0);
+        printf("EnumDecl\t\"");
+        print_cursor_spelling(cursor);
+        printf("\"\t\tParent = \"");
+        print_cursor_spelling(parent);
+        printf("\"\n");
+        clang_visitChildren(cursor, enum_child_visitor, 0);
 }
 
-void struct_handler(CXCursor cursor, CXCursor parent, CXClientData client_data) 
+void struct_handler(CXCursor cursor, CXCursor parent, CXClientData client_data)
 {
-  printf("StructDecl\t\"");
-  print_cursor_spelling(cursor);
-  printf("\"\t\tParent = \"");
-  print_cursor_spelling(parent);
-  printf("\"\n");
-  clang_visitChildren(cursor, struct_child_visitor, 0);
+        printf("StructDecl\t\"");
+        print_cursor_spelling(cursor);
+        printf("\"\t\tParent = \"");
+        print_cursor_spelling(parent);
+        printf("\"\n");
+        clang_visitChildren(cursor, struct_child_visitor, 0);
 }
 
-enum CXChildVisitResult enum_child_visitor(CXCursor cursor, CXCursor parent, CXClientData client_data) 
+void union_handler(CXCursor cursor, CXCursor parent, CXClientData client_data) 
 {
-  enum CXCursorKind cursor_kind = clang_getCursorKind(cursor);
-  switch(cursor_kind){
-    case CXCursor_IntegerLiteral: {
-      long long enum_const_value = clang_getEnumConstantDeclValue(parent);
-      printf("IntegerLiteral\t\t value = \"%lld\" \t\tParent = \"", enum_const_value);
-      print_cursor_spelling(parent);
-      printf("\"\n");
-      break;
-    }
-    
-    case CXCursor_EnumConstantDecl:
-      printf("EnumConstantDecl\t\"");
-      print_cursor_spelling(cursor);
-      printf("\"\t\tParent = \"");
-      print_cursor_spelling(parent);
-      printf("\"\n");
-      break;
-
-    default:
-      printf("Unaddressed - ");
-      print_CXString(clang_getCursorKindSpelling(cursor_kind));
-      printf("\n");
-  }
-  return CXChildVisit_Recurse;
+        printf("UnionDecl\t\"");
+        print_cursor_spelling(cursor);
+        printf("\"\t\tParent = \"");
+        print_cursor_spelling(parent);
+        printf("\"\n");
+        clang_visitChildren(cursor, struct_child_visitor, 0);
 }
 
-enum CXChildVisitResult struct_child_visitor(CXCursor cursor, CXCursor parent, CXClientData client_data)
+void function_handler(CXCursor cursor, CXCursor parent, CXClientData client_data)
+{
+        CXType type = clang_getCursorType(cursor);
+        int num_args = clang_Cursor_getNumArguments(cursor);
+        CINDEX_LINKAGE unsigned is_function_inline;
```

Comment:
This could be just a bool from stdbool.h.

Reply:
I tried that too. Did this because there is no need to include stdbool.h. I will change